internal/services/extendedlocation/extended_location_custom_location.go

```diff
@@ -0,0 +1,327 @@
+			client := metadata.Client.ExtendedLocation.CustomLocationsClient
+			id, err := customlocations.ParseCustomLocationID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("reading %s: %+v", *id, err)
+			}
+
+			if model := resp.Model; model != nil {
+				props := model.Properties
+
+				state := CustomLocationResourceModel{
+					Name:              id.CustomLocationName,
+					ResourceGroupName: id.ResourceGroupName,
+					Location:          model.Location,
+				}
+
+				if props.ClusterExtensionIds != nil {
+					state.ClusterExtensionIds = pointer.From(props.ClusterExtensionIds)
+				}
+				if props.DisplayName != nil {
+					state.DisplayName = pointer.From(props.DisplayName)
+				}
+				if props.HostResourceId != nil {
+					state.HostResourceId = pointer.From(props.HostResourceId)
+				}
+				if props.HostType != nil {
+					state.HostType = string(pointer.From(props.HostType))
+				}
+				if props.Namespace != nil {
+					state.Namespace = pointer.From(props.Namespace)
+				}
+
+				if props.Authentication != nil && props.Authentication.Type != nil && props.Authentication.Value != nil {
+					state.Authentication = []AuthModel{
+						{
+							Type:  pointer.From(props.Authentication.Type),
+							Value: pointer.From(props.Authentication.Value),
+						},
+					}
+				}
+
+				return metadata.Encode(&state)
+			}
+			return nil
+		},
+	}
+}
+
+func (r CustomLocationResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ExtendedLocation.CustomLocationsClient
+			id, err := customlocations.ParseCustomLocationID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			metadata.Logger.Infof("deleting %s", *id)
+
+			if resp, err := client.Delete(ctx, *id); err != nil {
+				if !response.WasNotFound(resp.HttpResponse) {
+					return fmt.Errorf("deleting %s: %+v", *id, err)
+				}
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r CustomLocationResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ExtendedLocation.CustomLocationsClient
+			id, err := customlocations.ParseCustomLocationID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var state CustomLocationResourceModel
+			if err := metadata.Decode(&state); err != nil {
+				return fmt.Errorf("decoding %+v", err)
+			}
+
+			customLocationProps := customlocations.CustomLocationProperties{}
+			d := metadata.ResourceData
+
+			if d.HasChanges("authentication") {
+				if len(state.Authentication) > 0 {
```

Comment:
If the user removes the `authentication` block, this won't remove the configuration from the resource as we're only processing changed blocks. Is this intentional? i.e. can this config item be removed once set? If not, this needs noting in the docs.

Reply:
My bad, it's now back.